Find a functionally equivalent code:android.animation.ValueAnimator.setDuration:COMMENT
Method Modifier: public      
Comment:/**
 * Sets the length of the animation. The default duration is 300 milliseconds.
 *
 * @param duration The length of the animation, in milliseconds. This value cannot
 * be negative.
 * @return ValueAnimator The object called with setDuration(). This return
 * value makes it easier to compose statements together that construct and then set the
 * duration, as in <code>ValueAnimator.ofInt(0, 10).setDuration(500).start()</code>.
 */

@@ -3,8 +3,8 @@
         throw new IllegalArgumentException("Animators cannot have negative duration: " + duration);
     }
     mUnscaledDuration = duration;
-    mDuration = (long) (duration * sDurationScale);
+    updateScaledDuration();
     return this;
 }

